Content.Server/Supermatter/EntitySystems/SupermatterSystem.cs

```diff
@@ -0,0 +1,425 @@
+
+                    float TempFactor;
+                    if (SMcomponent.GasmixPowerRatio > 0.8)
+                    {
+                        //with a perfect gas mix, make the power more based on heat
+                        TempFactor = 50f;
+                        //glow harder
+                    }
+                    else
+                    {
+                        //in normal mode, power is less effected by heat
+                        TempFactor = 30f;
+                        //glow normally
+                    }
+
+                    //power is set for radiation
+                    SMcomponent.Power = Math.Max(((temp * TempFactor / Atmospherics.T0C) * SMcomponent.GasmixPowerRatio) + SMcomponent.Power, 0);
+
+                    //more math to actually calculate radiation output
+                    _radiationSystem.Radiate(SMcomponent.Power * Math.Max(0f, (1f + (SMcomponent.PowerTransmissionBonus/10f))), 10f, SMcomponent, frameTime);
+
+                    //TODO: PsyCoeff should change from 0-1 based on psycologist distance
+                    float energy = SMcomponent.Power * SupermatterComponent.ReactionPowerModefier * (1f - (SMcomponent.PsyCoeff * 0.2f));
+
+                    //To figure out how much temperature to add each tick, consider that at one atmosphere's worth
+                    //of pure oxygen, with all four lasers firing at standard energy and no N2 present, at room temperature
+                    //that the device energy is around 2140. At that stage, we don't want too much heat to be put out
+                    //Since the core is effectively "cold"
+
+                    //Also keep in mind we are only adding this temperature to (efficiency)% of the one tile the rock
+                    //is on. An increase of 4*C @ 25% efficiency here results in an increase of 1*C / (#tilesincore) overall.
+                    //Power * 0.55 * (some value between 1.5 and 23) / 5
+
+                    temp += ((energy * SMcomponent.DynamicHeatModifier) / Atmospherics.ThermalReleaseModifier);
+
+                    //We can only emit so much heat, that being 57500
+                    temp = Math.Max(0, Math.Min(temp, 2500 * SMcomponent.DynamicHeatModifier));
+
+                    //Calculate how much gas to release
+                    //Varies based on power and gas content
+                    Pla += Math.Max(((energy * SMcomponent.DynamicHeatModifier) / Atmospherics.PlasmaReleaseModifier), 0f);
+                    //Varies based on power, gas content, and heat
+                    Oxy += Math.Max((((energy + temp * SMcomponent.DynamicHeatModifier) - Atmospherics.T0C) / Atmospherics.OxygenReleaseModifier), 0f);
+
+                    //i honestly have no idea what this is doing
+                    float FuckMeUp = (float) Math.Pow(SMcomponent.Power / 500f, 3f);
+
+                    //for supermatter soothing
+                    float PsyEffect = (1f - (0.2f * SMcomponent.PsyCoeff));
```

Comment:
isnt this just for if there's a psychologist nearby? we dont have that so this is kinda a pointless calc